### ISSUE TITLE
Serialization of nested classes

### ISSUE THREAD

**Issue by logicfish** [huntlabs/hunt#63]
```
unittest {
	class myClass {
		class myInner {
			
		}
		myInner[] innerArray;
	}
	myClass c = new myClass;
	auto s = serialize(c);
}
```
`\AppData\Local\dub\packages\hunt-1.3.0\hunt\source\hunt\util\Serialize.d(666,9): Error: outer function context of `app.__unittest_L24_C1` is needed to `new` nested class `app.__unittest_L24_C1.myClass``

**Comment by Heromyth** [comment]
A new binary serialization module named BinarySerializer is on the todo list.

**Comment by Heromyth** [comment]
It's fixed in commit 032fe2c.
